pkgs/development/python-modules/python-imports-check-libstdcxx/test_forward_compatibility_fails.py

```diff
@@ -0,0 +1,39 @@
+import os
+import sys
+
+assert (
+    "EXPECTED_LIBSTDCXX" in os.environ
+), "The derivation failed to export EXPECTED_LIBSTDCXX"
+
+expected_libstdcxx_root = os.environ["EXPECTED_LIBSTDCXX"]
+
+import load_libstdcxx_old
+import psutil
+
+try:
+    import load_libstdcxx_new
+except ImportError:
+    print(f"python-imports-check-load-libstdcxx successfuly fails", file=sys.stderr)
+    sys.exit(0)
+print(
+    "python-imports-check-load-libstdcxx doesn't fail the naive forward"
+    " compatibility test, it probably needs to be updated.",
+    file=sys.stderr,
+)
+
+loaded_libraries = [x.path for x in psutil.Process(os.getpid()).memory_maps()]
+loaded_libraries = [x for x in loaded_libraries if "libstdc++" in x]
+
+assert loaded_libraries, "Didn't load either version of libstdc++"
+
+assert all(x.startswith(expected_libstdcxx_root) for x in loaded_libraries), (
```

Comment:
IIUC this code path has already failed the test, so it seems a bit awkward to have `assert`s here. what is the purpose of `assert` if we already failed the test?

Reply:
It checks if the test might have failed for an unforeseen reason. Might be handy when the test gets out of date. Or it might be entirely useless